src/main/kotlin/com/doist/gradle/androidtranslationscheck/TranslationFileBuilder.kt

```diff
@@ -0,0 +1,69 @@
+package com.doist.gradle.androidtranslationscheck
+
+/**
+ * Builds [TranslationFile] from the given [String].
+ */
+internal class TranslationFileBuilder(
+    private val regexArg: Regex
+) {
+    fun buildFrom(text: String): TranslationFile {
+        return TranslationFile(
+            text.findTranslationStrings(),
+            text.findTranslationStringArrays(),
+            text.findPlurals()
+        )
+    }
+
+    private fun String.findTranslationStrings(): Set<TranslationString> {
+        return regexString.findAll(this).mapTo(mutableSetOf()) {
+            val (name, string) = it.destructured
+            TranslationString(name, string.findArguments())
+        }
+    }
+
+    private fun String.findTranslationStringArrays(): Set<TranslationStringArray> {
+        return regexStringArray.findAll(this).mapTo(mutableSetOf()) { stringArray ->
```

Comment:
Nit. Other `find` methods have `it` instead of `stringArray`. Let's make it consistent?